actioncable.go

```diff
@@ -94,6 +97,44 @@ func (c *Client) Subscribe(channel string) (<-chan *EventOrErr, error) {
 	}
 
 	c.subscriptions[channel] = make(chan *EventOrErr, 32)
+	c.subscData[channel], err = json.Marshal(&innerIdentifier{Channel: channel})
+	if err != nil {
+		panic(err)
+	}
+
+	// Add subscription command to outbound queue.
+	go func() {
+		c.subc <- channel
+	}()
+
+	return c.subscriptions[channel], nil
+}
+
+var ErrNoChannelName = errors.New("Subscription data must contain a channel name")
+
+// SubscribeWith subscribes to a channel giving parameters
+// the params must contain a field "channel" containing the channe to subscribe to
+func (c *Client) SubscribeWith(params json.RawMessage) (<-chan *EventOrErr, error) {
```

Comment:
yeah you're right, this is made quite difficult by the fact that the arbitrary JSON is merged in at the top level. Go just doesn't make that easy =/

I've thought this through a bit and I'm still leaning towards the `map[string]interface{}` solution as being cleaner overall, but I'm definitely conflicted about it.

Maybe we can still ask for `channel` as an explicit argument before the `json.RawMessage`/`[]byte`? Does that avoid us having to do anything inside this function to attempt to decipher or manipulate the payload? It can of course come with the caveat clearly called out that the payload needs to include a `channel` field.